### PR TITLE
Imagery providers were incorrectly created for WMTS and AzureMaps. Improved testing.

### PR DESCRIPTION
**Pull request by mdastous-bentley**

### CODE REVIEW

File: core/frontend/src/test/tile/map/MapLayerImageryFormats.test.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    internalMapLayerImageryFormats.forEach((imageryFormat) => {
+
+      const layerSettings = getSampleLayerSettings(imageryFormat.formatId);
+      expect(layerSettings).to.not.undefined;
+      if (!layerSettings)
+        return;
+      const provider = registry.createImageryProvider(layerSettings);
+
+      expect(provider).to.not.undefined;
+      if (!layerSettings)
```
Comment (pmconne) [inline core/frontend/src/test/tile/map/MapLayerImageryFormats.test.ts:38]: `/s/layerSettings/provider` ?

Comment (mdastous-bentley) [reply core/frontend/src/test/tile/map/MapLayerImageryFormats.test.ts:38]: Actually no check is needed at all after creating the provider, I removed it.

**Review by pmconne — APPROVED** [review]

File: core/frontend/src/test/tile/map/MapLayerImageryFormats.test.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          break;
+
+        case "MapboxImagery":
+          expect(provider instanceof MapBoxLayerImageryProvider).to.true;
+          break;
+
+        case "TileUrlMapLayerFormat":
+          expect(provider instanceof TileUrlImageryProvider).to.true;
+          break;
+      }
```
Comment (mdastous-bentley) [inline core/frontend/src/test/tile/map/MapLayerImageryFormats.test.ts:64]: @pmconne Actually I think this test should fail if a new format is added and is not covered by this test.  What do you think?

Comment (pmconne) [reply core/frontend/src/test/tile/map/MapLayerImageryFormats.test.ts:64]: Yes.

**Pull request merged**